Upgrade to latest fabric go mod packages.

SamYuan1990 opened the pull request:
Nil pointer handler
Upgrade to latest fabric go mod packages.
Updated CI

SamYuan1990 commented:
@guoger , do we need to change ci to enable go mod?

```
# Go
# Build your Go project.
# Add steps that test, save build artifacts, deploy, and more:
# https://docs.microsoft.com/azure/devops/pipelines/languages/go

trigger:
- master

pool:
  vmImage: 'ubuntu-latest'

variables:
  GOBIN:  '$(GOPATH)/bin' # Go binaries path
  GOROOT: '/usr/local/go1.11' # Go installation path
  GOPATH: '$(system.defaultWorkingDirectory)/gopath' # Go workspace path
  GO111MODULE: 'on'
  modulePath: '$(GOPATH)/src/github.com/$(build.repository.name)' # Path to the module's code

steps:
- script: |
    mkdir -p '$(GOBIN)'
    mkdir -p '$(GOPATH)/pkg'
    mkdir -p '$(modulePath)'
    shopt -s extglob
    shopt -s dotglob
    mv !(gopath) '$(modulePath)'
    echo '##vso[task.prependpath]$(GOBIN)'
    echo '##vso[task.prependpath]$(GOROOT)/bin'
  displayName: 'Set up the Go workspace'

- script: |
    go version
    go mod tidy
    go test ./test -test.bench=".*"
    go build -v .
  workingDirectory: '$(modulePath)'
  displayName: 'Get dependencies, then build'
```

guoger commented:
It seems that mock fabric from SDK does not actually validate anything, and by sending tx to mock peer, we are merely testing the grpc connection, which is not super helpful IMO.

could we consider #15 instead?

SamYuan1990 commented:
> #15

UT as unit test to keep code works. #15 looks like sample of the project. In my opinion, diff things.

SamYuan1990 commented:
@guoger , just rebase with latest master, pls take a review.

SamYuan1990 commented:
@guoger , just rebase with latest master, pls take a review.

SamYuan1990 commented:
currently cover function
```
proposer.CreateProposer
proposer.Start
```

guoger commented:
pls split the part you bump fabric version (and switch to use new proto pkg) to a **separate PR** so we could get that merged quickly

SamYuan1990 commented:
> pls split the part you bump fabric version (and switch to use new proto pkg) to a **separate PR** so we could get that merged quickly

removed test file, changed pr as Upgrade to latest fabric go mod packages.

guoger commented on go.mod:
```diff
@@ -3,22 +3,24 @@ module github.com/guoger/stupid
 	github.com/hashicorp/go-version v1.2.0 // indirect
-	github.com/hyperledger/fabric v1.4.3
-	github.com/hyperledger/fabric-amcl v0.0.0-20181230093703-5ccba6eab8d6 // indirect
+	github.com/hyperledger/fabric v2.0.1+incompatible
```
fabric internalized `comm` pkg in later releases. I think we could actually remove fabric from dependencies entirely (copy some code over)

SamYuan1990 commented:
address the comment in separate PR pls.

guoger commented:
@SamYuan1990 i've enabled sign-off check in CI. Please make sure to use `git commit -s` for future contributions (and there should a blank line before sign-off line)
